src/main/webapp/app/complaints-for-tutor/complaints-for-tutor.component.ts

```diff
@@ -4,48 +4,31 @@ import { ComplaintService } from 'app/entities/complaint/complaint.service';
 import { HttpErrorResponse } from '@angular/common/http';
 import { ComplaintResponseService } from 'app/entities/complaint-response/complaint-response.service';
 import { ComplaintResponse } from 'app/entities/complaint-response';
-import { Complaint } from 'app/entities/complaint';
+import { Complaint, ComplaintType } from 'app/entities/complaint';
 
 @Component({
     selector: 'jhi-complaints-for-tutor-form',
     templateUrl: './complaints-for-tutor.component.html',
     providers: [JhiAlertService],
 })
 export class ComplaintsForTutorComponent implements OnInit {
-    @Input() resultId: number;
+    @Input() complaint: Complaint;
     @Input() isAllowedToRespond: boolean; // indicates if the tutor is allowed to respond (i.e. that he is not the assessor)
     // Indicates that the assessment should be updated after a complaint. Includes the corresponding complaint
     // that should be sent to the server along with the assessment update.
     @Output() updateAssessmentAfterComplaint = new EventEmitter<ComplaintResponse>();
-    loading = true;
-    complaint: Complaint;
     complaintText = '';
-    alreadySubmitted: boolean;
     handled: boolean;
     complaintResponse: ComplaintResponse = new ComplaintResponse();
 
     constructor(private complaintService: ComplaintService, private jhiAlertService: JhiAlertService, private complaintResponseService: ComplaintResponseService) {}
 
     ngOnInit(): void {
-        this.complaintService.findByResultId(this.resultId).subscribe(
-            res => {
-                if (!res.body) {
-                    return;
-                }
-                this.complaint = res.body;
-                this.complaintText = this.complaint.complaintText;
-                this.handled = this.complaint.accepted !== undefined;
-                this.alreadySubmitted = true;
-                this.loading = false;
-
-                if (this.handled) {
-                    this.complaintResponseService.findByComplaintId(res.body.id).subscribe(complaintResponse => (this.complaintResponse = complaintResponse.body!));
-                }
-            },
-            (err: HttpErrorResponse) => {
-                this.onError(err.message);
-            },
-        );
+        this.complaintText = this.complaint.complaintText;
```

Comment:
The component `modeling-assessment-editor` uses this component, without inputting a complaint.
This should immediately cause a console error.
Also `modeling-assessment-editor` passes in a result id, which is not supported anymore.